tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -1519,6 +1519,47 @@ TEST_F(DataserviceReadVersionedLayerClientTest,
   }
 }
 
+TEST_F(DataserviceReadVersionedLayerClientTest, PrefetchAggregatedTile) {
+  constexpr auto kLayerId = "hype-test-prefetch";
+
+  auto client = std::make_shared<read::VersionedLayerClient>(
+      kCatalog, kLayerId, boost::none, settings_);
+
+  {
+    SCOPED_TRACE("Prefetch aggregated tile");
+    std::vector<geo::TileKey> tile_keys = {
+        geo::TileKey::FromHereTile("23618365")};
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(URL_QUADKEYS_92259), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(GetResponse(http::HttpStatusCode::OK),
+                                     HTTP_RESPONSE_QUADKEYS_92259_ROOT_ONLY));
+
+    auto request = read::PrefetchTilesRequest()
+                       .WithTileKeys(tile_keys)
+                       .WithDataAggregationEnabled(true);
+
+    auto promise = std::make_shared<std::promise<PrefetchTilesResponse>>();
+    auto future = promise->get_future();
+    auto token = client->PrefetchTiles(
+        request, [promise](PrefetchTilesResponse response) {
+          promise->set_value(std::move(response));
+        });
+
+    ASSERT_NE(future.wait_for(kWaitTimeout), std::future_status::timeout);
+    PrefetchTilesResponse response = future.get();
+    ASSERT_TRUE(response.IsSuccessful()) << response.GetError().GetMessage();
+    ASSERT_FALSE(response.GetResult().empty());
+
+    const auto& result = response.GetResult();
+
+    for (auto tile_result : result) {
+      ASSERT_TRUE(tile_result->IsSuccessful());
+      ASSERT_TRUE(tile_result->tile_key_.IsValid());
+    }
```

Comment:
Why do we expect more than 1 tile in result?
Also, maybe, we can create QuarTreeIndex from response and compare  PrefetchTiles() result with `QuadTreeIndex::Find(tile, true)`.

Reply:
It will be more 'challenging' if quad tree has more tiles :-)

Reply:
Change the condition